### ISSUE TITLE
calcuts "mean is not significantly different with peak"

### ISSUE THREAD

**Issue by cwb14** [dfguan/purge_dups#30]
I have PacBio Hifi data for a diploid plant species (1.1% heterozygosity). I assembled the genome using HiCanu and aligned raw reads back to the genome using:
`minimap2 -t 20 -I 16G -x HiCanu.contigs.fasta ~/scratch/supina/reads.ccs.fa | gzip -c - > minimap2.paf.gz`
calcuts on that alignment generates the following error message:
`[M::calcuts] Merge local peaks and valleys: 0 peaks remain
[M::calcuts] Remove peaks and valleys less than 5: 0 peaks remain
[M::calcuts] The valley is not in proper position, use one-peak mode
[M::calcuts] mean: 29, peak: 32, mean not larger than peak, treat as haploid assembly
[W::calcuts] mean is not significantly different with peak, please recheck the cutoffs`
hist_plot.py generated the attached read depth histogram.
![test](https://user-images.githubusercontent.com/35149326/78298746-eaebe780-7500-11ea-895e-5d77c4e9ebda.png)
Should I try to manually assign cutoffs at 15 and 55, or does the one-peaked histogram suggest that I should tweak an alignment parameter?

**Comment by dfguan** [comment]
Hi, I can not see which parameter you are using for alignment, but it should be `-xasm20` suggested by Hengli here https://github.com/lh3/minimap2/issues/325. The high cutoff (90) is a little bit small, may try 160 (2.5 * 64). May I ask why so many bases have very low coverage? Is it due to the alignment? Best, Dengfeng.

**Comment by cwb14** [comment]
Hi, Dengfeng. Thanks for the quick response.

The plot above was generated using `-x map-pb`. I just redid the minimap2 alignment using `-x asm20`.  The curve was nearly identical, but using `asm20` had a higher initial peak with many bases containing very low coverage. Im not sure why that initial spiked peak exists... whether its
 a product of HiFi reads, the assembly, or alignment. 

Here is the `-x asm20` alignment with the cutoffs that you recommended. Thanks for your help!

![sup_asm20_cutoffs](https://user-images.githubusercontent.com/35149326/78459510-7d1df800-7687-11ea-9ff6-bfac7426540f.png)

**Comment by dfguan** [comment]
Yeah, those cutoffs look ok to me, you could try with purge_dups, and see what results you can get. Except the low coverage peak is still there, so weird, not sure what is happening. After purging, you can recalculate coverage histogram and the busco scores to validate the results. You can also use HIFI reads to make a KAT plot. Any questions, please feel free to ask. Dengfeng.

**Comment by cwb14** [comment]
Good ideas! And thanks for the continued feedback. Here are the busco scores for the original HiCanu-generated genome `C:97.8%[S:30.5%,D:67.3%],F:0.4%,M:1.8%,n:1614`.

After purging, the busco scores are `C:90.7%[S:87.1%,D:3.6%],F:0.9%,M:8.4%,n:1614`. Fantastic reduction in duplicate genes! But looks like we lost about 7% of complete genes. I also mapped the raw HIFI reads to the purged.fa genome and generated the following histogram. `Calcuts` identified the cutoffs. Do you have any advice for how I might be able to recover some of the 7% of buscos that I lost after purging?
![purged_asm20_cutoffs](https://user-images.githubusercontent.com/35149326/78522653-1e966e00-779c-11ea-96f5-6a2e7c5a8011.png)

**Comment by dfguan** [comment]
Hi, maybe you could try to use a smaller middle coverage, a value in 40~54. I have very little experience in plant assembly. Dengfeng.

**Comment by cwb14** [comment]
Thank you! I will give this a shot. I also had a bit of luck adjusting the minimum alignment score (`-a`) of the `purge_dups` function.  Default is `-a 70` but when I adjusted to `-a 80`, my purged BUSCOs were `C:95.9%[S:90.4%,D:5.5%],F:0.9%,M:3.2%`.

I borrowed this idea from https://bitbucket.org/mroachawri/purge_haplotigs/issues/60/lower-busco.

**Comment by bishopia** [comment]
> Hi, I can not see which parameter you are using for alignment, but it should be `-xasm20` suggested by Hengli here [lh3/minimap2#325](https://github.com/lh3/minimap2/issues/325). The high cutoff (90) is a little bit small, may try 160 (2.5 * 64). May I ask why so many bases have very low coverage? Is it due to the alignment? Best, Dengfeng.

Just to check after some more updates in minimap2: if i have hifi reads, for purge_dups step 1 paf generation, it would be best to use ```-map-hifi```, not ```-map-pb``` as in README or ```-xasm20``` as above, correct?

**Comment by gushiro** [comment]
> Hi, I can not see which parameter you are using for alignment, but it should be `-xasm20` suggested by Hengli here [lh3/minimap2#325](https://github.com/lh3/minimap2/issues/325). The high cutoff (90) is a little bit small, may try 160 (2.5 * 64). May I ask why so many bases have very low coverage? Is it due to the alignment? Best, Dengfeng.

@dfguan why you suggest 2.5*64 ? I assume 64 is the peak with the primary contigs, but why 2.5 if I might ask? Thanks in advance
